src/GeneralStateTestsFiller/stEWASMTests/README.md

```diff
@@ -0,0 +1,317 @@
+
+Then you have to define the memory to be used (all tests right now uses 1):
+
+```lisp
+(memory 1)
+```
+
+then you have to export the memory and the function main:
+
+```lisp
+(export "memory" (memory 0))
+(export "main" (func $main))
+```
+
+finally define the main function:
+
+```lisp
+(func $main
+  (i32.store (i32.const 0) (call $getCallDataSize))
+  (call $storageStore (i32.const 100) (i32.const 0))
+)
+```
+
+#### debugging
+
+Hera also provides options to [debug wast code][14]. In order to use this debug
+option we need to build hera with the flag `-DHERA_DEBUGGING=ON`.
+
+The most common used method for debugging is `printMemHex`, this methods allows
+us to print parts of the WASM memory, this method receives two parameters:
+`offset` and `length`, this print the number of bytes specified in `length`
+starting at `offset`.
+
+After you have debugged your test case and you are sure it is working as
+expected you should remove the debug imports and statements, this is because
+when Hera is built without the `-DHERA_DEBUGGING=ON` flag all tests which
+includes debug statements will fail.
+
+- How to import a debug method:
+
+```lisp
+(import "debug" "printMemHex" (func $printMemHex (param i32 i32)))
+```
+
+- How to call a debug method:
+
+```lisp
+(call $printMemHex (i32.const 0) (i32.const 32))
+```
+
+### 3. expect - expected result
+
+Example:
+
+```yaml
+expect:
+  - indexes:
+    data: !!int -1
+    gas: !!int -1
+    value: !!int -1
+  network:
+    - ALL
+  result:
+    # InitialBalance - contract1.call - valueTransfer - txCost - contract2.calldatasize - contract2.sstore
+    # 100000000000   - 700            - 9000          - 21000  - 2                      - 20000            = 99999949298
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '99999949298'
+    deadbeef00000000000000000000000000000000:
+      storage: {
+      }
+    deadbeef00000000000000000000000000000001:
+      storage: {
+        0: "0x4"
+      }
+      balance: "100000000001"
+```
+
+For the expect section values of indexes and network remain the same in all test
+cases, the most important field in this section is `result`.
+
+In `result` field we specify the accounts as we do in the `pre` section, but now
+we define the values that we expect as result of the transaction and the
+execution of the wast code.
+
+```yaml
+deadbeef00000000000000000000000000000001:
+   storage: {
+     0: "0x4"
+   }
+   balance: "100000000001"
+```
+### 4. transaction
+
+Transaction contains the following fields:
+
+- data
+- gasLimit
+- gasPrice - usually set to 1 to make it easy to calculate gas
+- nonce
+- secretKey - using
+  `45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8` which is
+  the secret key for account
+  `45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8`
```

Comment:
This seems to be a private key not an address?

Reply:
good catch